src/common/fork.c

```diff
@@ -33,12 +35,27 @@ forkDetach(void)
 {
     FUNCTION_LOG_VOID(logLevelTrace);
 
-    // Change the working directory to / so we won't error if the old working directory goes away
-    THROW_ON_SYS_ERROR(chdir("/") == -1, PathMissingError, "unable to change directory to '/'");
-
     // Make this process a group leader so the parent process won't block waiting for it to finish
     THROW_ON_SYS_ERROR(setsid() == -1, KernelError, "unable to create new session group");
 
+    // The process should never receive a SIGHUP but ignore it just in case
+    signal(SIGHUP, SIG_IGN);
+
+    // There should be no way the child process can exit first (after the next fork) but just in case ignore SIGCHLD.  This means
+    // that the child process will automatically be reaped by the kernel should it finish first rather than becoming defunct.
+    signal(SIGCHLD, SIG_IGN);
+
+    // Fork again and let the parent process terminate to ensure that we get rid of the session leading process. Only session
+    // leaders may get a TTY again.
+    if (forkSafe() > 0)
```

Comment:
This is probably fine but it made me think for a bit about if forkSafe() could possibly return a less-than-zero value.  Seems like it might draw fewer questions to make this the exact opposite of the regular if (forkSafe() == 0) check, ie:

 if (forkSafe() != 0)

Instead, since forkSafe() is defined to return 0 for child, non-zero for parent, and throw an exception on error.  In other words, the interface for forkSafe() isn't quite the same as that of fork() itself.  Anyway, relatively minor point, either way should work.

Reply:
Works for me.